### ISSUE TITLE
`mandatory_comma: true` incorrectly indents the next line at end of literals.

### ISSUE THREAD

**Issue by j-h-a** [realm/SwiftLint#2226]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

The `trailing_comma` rule with `mandatory_comma: true` does not behave correctly with the `autocorrect --format` option, incorrectly aligning the next line at the end of literals. This makes it incompatible to use with the `literal_expression_end_indentation` opt-in rule.

### Environment

* SwiftLint version: `0.25.1`
* Installation method used: `CocoaPods`

**Minimal triggering example:**

`.swiftlint.yml`:
```yml
opt_in_rules:
   - literal_expression_end_indentation
trailing_comma:
  mandatory_comma: true
```

`example.swift`:
```swift
struct TheStruct {
    let anArray = [
        "a",
        "b"
    ]
}
```

To test this example create the above two files and run `swiftlint autocorrect --format`

**Expected behaviour** (`example.swift` after autocorrect):
```swift
struct TheStruct {
    let anArray = [
        "a",
        "b",
    ]
}
```
Comma is inserted after `"b"` and square bracket on line below is left alone.

**Actual behaviour** (`example.swift` after autocorrect):
```swift
struct TheStruct {
    let anArray = [
        "a",
        "b",
        ]
}
```
Comma is inserted after `"b"` and square bracket on line below is indented to the same level as the array element.

A subsequent `swiftlint lint --path example.swift --no-cache` command will now trigger a warning for the "Literal Expression End Indentation Violation".

**Comment by marcelofabri** [comment]
It looks like the problem is not the trailing_comma rule, but `--format`. `--format` uses SourceKit to format the file, which probably is formatting in the same style as Xcode.

**Comment by j-h-a** [comment]
I see - I wasn't aware that the autocorrect/formatting was done by a separate tool. This is only triggered when the `trailing_comma` rule is enabled, so does that mean SourceKit indentation is being triggered based on the comma from the previous line? I've found the SourceKit repo, but not really sure how to proceed - it has GH issues disabled. Any advice?

**Comment by marcelofabri** [comment]
You can try to use [SourceKitten](https://github.com/jpsim/SourceKitten) to call just `format` to see what happens.

I'm not sure it's a bug on SourceKit, it may just be the expected (although ugly) behavior. SourceKit is part of the Swift toolchain, so bugs should be reported to https://bugs.swift.org.

**Comment by j-h-a** [comment]
Confirmed: `sourcekitten format --file example.swift` is causing this behaviour. Bracket is indented when there is a comma and not when there isn't one.

**Comment by marcelofabri** [comment]
I'm closing this issue then. Thanks for the update!
